fuel/src/main/kotlin/com/github/kittinunf/fuel/core/Headers.kt

```diff
@@ -49,44 +63,52 @@ class Headers : MutableMap<String, HeaderValues> {
     /**
      * Associates the specified [value] with the specified [key] in the map.
      *
-     * @return the previous value associated with the key, or `null` if the key was not present in the map.
+     * @param key [String] the header name
+     * @param value [HeaderValues] the new values
+     * @return [HeaderValues?] the previous value at [key], or `null` if the [key] was not present in the map.
      */
     override fun put(key: String, value: HeaderValues): HeaderValues? {
         return contents.put(HeaderName(key), value)
     }
 
-    fun append(key: String, value: Collection<*>): Headers {
-        val current = contents.getOrPut(HeaderName(key)) { emptyList() }
-        put(key, current.plus(value.map { it.toString() }))
+    /**
+     * Appends the entire [values] to the current values at [header]
+     *
+     * @param header [String] the header to append to
+     * @param values [Collection] the values to append, coerced with [Any.toString]
+     */
+    fun append(header: String, values: Collection<*>): Headers {
+        put(header, this[header].plus(values.map { it.toString() }))
         return this
     }
 
-    fun append(key: String, value: Any): Headers {
-        // Some headers can not be appended per the RFC
-        if (Headers.isSingleValue(HeaderName(key))) {
-            set(key, value.toString())
-            return this
+    /**
+     * Appends a single value [value] to the current values at [header]
+     *
+     * @param header [String] the header to append to
+     * @param value [Any] the value to append, coerced with [Any.toString]
+     */
+    fun append(header: String, value: Any): Headers {
+        return when (Headers.isSingleValue(header)) {
+            true -> set(header, value.toString())
+            false -> set(header, this[header].plus(value.toString()))
         }
```

Comment:
would you consider 

``` kotlin
fun append(header: String, value: Any) = set(header, if(Headers.isSingleValue(header)value.toString() else this[header].plus(value.toString())))
        
```

Reply:
To me that looks horrible, but if that's our coding style, sure!

Reply:
``` Kotlin
fun append(header: String, value: Any): Headers {
  val value = if(Headers.isSingleValue(header)) value.toString() else this[header].plus(value.toString())
  return set(header, value)
}
```